File: Searching/Binary Search/BinarySearch.py
#iertative binary search v1

def binary_search(arr,low,high,target):

    while low<=high:
        mid = (high + low) // 2
        if arr[mid] < target: 
            low = mid + 1
        elif arr[mid] > target: 
            high = mid - 1
        else: 
            return mid
    
    return -1




if __name__=="__main__":

    n = int(input("Number of elements: "))
    arr = list(map(int, input("Enter the elements: \n").split()))
    target = int(input("Enter element to be found: "))

    arr.sort()

    print("Sorted elements: {}".format(arr))
    index = binary_search(arr, 0, n-1, target)

    if index==-1:
        print("Element not found.")
    else:
        print("Element found at position: {}".format(index))

File: Searching/Binary Search/binarySearch.c
#include <stdio.h> 
  
int binarySearch(int arr[], int left, int right, int x) 
{ 
    while (left <= right) { 
        int mid = left + (right - l) / 2; 
  
        if (arr[mid] > x) 
            right = mid - 1;
  
        if (arr[mid] < x) 
            left = mid + 1; 
  
        else
            return mid;
             
    } 
  
    return -1; 
} 
  
int main(void) 
{ 
    int arr[] = { 2, 3, 4, 10, 40 }; 
    int n = sizeof(arr) / sizeof(arr[0]); 
    int x = 10; 
    int result = binarySearch(arr, 0, n - 1, x); 
    (result == -1) ? printf("Element is not present"
                            " in array") 
                   : printf("Element is present at "
                            "index %d", 
                            result); 
    return 0; 
} 